docs/products/polygon-id-state-replication/tutorial-verifying-polygon-id-credentials-in-ethereum-dapp.mdx

```diff
@@ -1 +1,597 @@
+          tx: string
+        }>(`${config.RARIMO_CORE_API_URL}/integrations/relayer/state/relay`, {
+          body: {
+            hash: claimStateHex,
+            chain: RELAYER_RELAY_CHAIN_NAMES[config.DEFAULT_CHAIN],
+          },
+        })
+
+        if (!data?.tx) throw new Error('tx is not defined')
+
+        await waitTx(data?.tx)
+
+        return false
+      } catch (error) {
+        return handleStateValidatingError(error)
+      }
+    },
+    [handleStateValidatingError, waitTx],
+  )
+
+  const isGistStateValid = useCallback(
+    async (gistStateHash: string) => {
+      try {
+        const { data } = await fetcher.post<{
+          tx: string
+        }>(`${config.RARIMO_CORE_API_URL}/integrations/relayer/gist/relay`, {
+          body: {
+            hash: gistStateHash,
+            chain: RELAYER_RELAY_CHAIN_NAMES[config.DEFAULT_CHAIN],
+          },
+        })
+
+        if (!data?.tx) throw new Error('tx is not defined')
+
+        await waitTx(data?.tx)
+
+        return false
+      } catch (error) {
+        return handleStateValidatingError(error)
+      }
+    },
+    [handleStateValidatingError, waitTx],
+  )
+```
+
+Finally, after transiting the state and getting the JWZ, we can submit the proof:
+
+```ts title="src/pages/AuthConfirmation/AuthConfirmation.tsx"
+/* ... */
+
+const submitZkp = useCallback(async () => {
+    setIsSubmitting(true)
+
+    try {
+        if (!jwzToken) throw new TypeError('ZKP is not defined')
+
+        const zkProofPayload = JSON.parse(jwzToken.getPayload())
+
+        const zkProof = zkProofPayload.body.scope[0] as ZKProof
+
+        const txBody = getProveIdentityTxBody(
+            '1',
+            zkProof.pub_signals.map(el => BigInt(el)),
+            [zkProof.proof.pi_a[0], zkProof.proof.pi_a[1]],
+            [
+            [zkProof.proof.pi_b[0][1], zkProof.proof.pi_b[0][0]],
+            [zkProof.proof.pi_b[1][1], zkProof.proof.pi_b[1][0]],
+            ],
+            [zkProof.proof.pi_c[0], zkProof.proof.pi_c[1]],
+        )
+
+        const tx = await provider?.signAndSendTx?.({
+            to: config?.[
+                `QUERY_VERIFIER_CONTRACT_ADDRESS_${selectedChainToPublish}`
+            ],
+            ...txBody,
+        })
+
+        verificationSuccessTx.set((tx as EthTransactionResponse).transactionHash)
+
+        navigate(RoutesPaths.authSuccess)
+        } catch (error) {
+        ErrorHandler.process(error)
+        }
+
+        setIsSubmitting(false)
+    }, [
+        getProveIdentityTxBody,
+        jwzToken,
+        navigate,
+        provider,
+        selectedChainToPublish,
+        verificationSuccessTx,
+    ])
+
+/* ... */
+```
+
+Now you're ready to use the Polygon ID State Replication with Rarimo!
+
+If you want to proceed further, we recommend you to clone the front-end from the [GitHub](https://github.com/rarimo/web-client-polygonid/tree/main), make the changes you need and use it for your purposes.
```

Comment:
Instead of "proceed further" - "tweak or expand this example", also add a link to contracts